SourceCode/Homework04/CRTTriangle.cpp

```diff
@@ -0,0 +1,37 @@
+#include "CRTTriangle.h"
+
+
+CRTTriangle::CRTTriangle(const CRTVector vertices[VERTICES]) {
+	for (int i = 0; i < VERTICES; i++) {
+		this->vertices[i] = vertices[i];
+	}
+}
+
+CRTTriangle::CRTTriangle(const CRTVector& v1, const CRTVector& v2, const CRTVector& v3) {
+	vertices[0] = v1;
+	vertices[1] = v2;
+	vertices[2] = v3;
+}
+
+/* the order of the vertices matters
+		v2
+
+
+	v0		v1
+*/
+CRTVector CRTTriangle::normalVector() const {
+	CRTVector v1 = vertices[1] - vertices[0];
+	CRTVector v2 = vertices[2] - vertices[0];
+
+	return cross(v2, v1); // we return the clockwise cross-product

```

Comment:
We've assume that we are using counterclockwise order, for now it's not a big difference but for the future tasks it could be wrong

Reply:
Is there a way to make sure we're taking the vertices in a counterclockwise order? Or should we do the calculation twice and return the one that points towards the camera?